src/Component/Filter/DateFilter.php

```diff
@@ -41,11 +41,14 @@ public function apply(DataSourceInterface $dataSource, string $name, $data, arra
     {
         $expressionBuilder = $dataSource->getExpressionBuilder();
 
-        $field = (string) $this->getOption($options, 'field', $name);
+        /** @var string $field */
+        $field = $options['field'] ?? $name;
```

Comment:
Do you still need to cast it to string, like it was before?

```suggestion
        $field = $options['field'] ? (string) $options['field'] : $name;
```

or the way how it's done bellow

```php
            /** @var bool|string|int $inclusive */
            $inclusive = $options['inclusive_from'] ?? self::DEFAULT_INCLUSIVE_FROM;
            $inclusive = (bool) $inclusive;
```

Reply:
I simplify the PHPDoc by defining the array shape on the method.
It allows to really understand what is expected here.